shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -894,6 +890,25 @@ protected FlutterEngine getFlutterEngine() {
     return delegate.getFlutterEngine();
   }
 
+  private Bundle cachedMetaData;
+
+  /** Mocks the meta data for testing purposes. */
+  @VisibleForTesting
+  public void setMetaData(Bundle metaData) {
```

Comment:
If your test is in the same package, you don't need public. Just `void setMetaData` which implies package private visibility. 

But also, don't add this method :) 

Just use https://javadoc.io/static/org.mockito/mockito-core/3.2.4/org/mockito/Mockito.html#spy-T- on getMetaData in tests (or just override it in a subclass for the class-under-test).